src/order/lattice_intervals.lean

```diff
@@ -32,106 +32,118 @@ namespace set
 
 namespace Ico
 
-variables {a b : α}
-
-instance [semilattice_inf α] : semilattice_inf (Ico a b) :=
+instance [semilattice_inf α] {a b : α} : semilattice_inf (Ico a b) :=
 subtype.semilattice_inf (λ x y hx hy, ⟨le_inf hx.1 hy.1, lt_of_le_of_lt inf_le_left hx.2⟩)
 
 /-- `Ico a b` has a bottom element whenever `a < b`. -/
-@[reducible] protected def order_bot [partial_order α] (h : a < b) : order_bot (Ico a b) :=
+@[reducible] protected def order_bot [partial_order α] {a b : α} (h : a < b) :
+  order_bot (Ico a b) :=
 (is_least_Ico h).order_bot
 
+@[simp] lemma coe_inf [semilattice_inf α] {a b : α} (x y : Ico a b) :
+  (↑(x ⊓ y) : α) = (↑x ⊓ ↑y) := rfl
+
 end Ico
 
 namespace Iio
 
 instance [semilattice_inf α] {a : α} : semilattice_inf (Iio a) :=
 subtype.semilattice_inf (λ x y hx hy, lt_of_le_of_lt inf_le_left hx)
 
+@[simp] lemma coe_inf [semilattice_inf α] {a : α} (x y : Iio a) : (↑(x ⊓ y) : α) = (↑x ⊓ ↑y) := rfl
+
 end Iio
 
 namespace Ioc
 
-variables {a b : α}
-
-instance [semilattice_sup α] : semilattice_sup (Ioc a b) :=
+instance [semilattice_sup α] {a b : α} : semilattice_sup (Ioc a b) :=
 subtype.semilattice_sup (λ x y hx hy, ⟨lt_of_lt_of_le hx.1 le_sup_left, sup_le hx.2 hy.2⟩)
 
 /-- `Ioc a b` has a top element whenever `a < b`. -/
-@[reducible] protected def order_top [partial_order α] (h : a < b) : order_top (Ioc a b) :=
+@[reducible] protected def order_top [partial_order α] {a b : α} (h : a < b) :
+  order_top (Ioc a b) :=
 (is_greatest_Ioc h).order_top
 
+@[simp] lemma coe_sup [semilattice_sup α] {a b : α} (x y : Ioc a b) :
+  (↑(x ⊔ y) : α) = (↑x ⊔ ↑y) := rfl
```

Comment:
You can be less verbose:
```suggestion
  (↑(x ⊔ y) : α) = x ⊔ y := rfl
```
Also, you probably should mark this lemma as `norm_cast`.
BTW, does it work if you add `subtype.coe_sup` as a `simp` lemma right after `subtype.semilattice_sup`?

Reply:
That last suggestion works - very nice!